### ISSUE TITLE
Wrong unicorn/regex-shorthand fix

### ISSUE THREAD

**Issue by ai** [sindresorhus/eslint-plugin-unicorn#477]
Code:

```js
const RE_AT_END = /[ \n\t\r\f"#'()/;[\\\]{}]/g
const RE_WORD_END = /[ \n\t\r\f(){}:;@!'"\\\][#]|\/(?=\*)/g
```

ESLint output:

```
/home/ai/Dev/postcss/lib/tokenize.js
  21:19  error  /[ \n\t\r\f{}()'"\\;/[\]#]/g can be optimized to /["#'()/;[\\\]{}]/g                        unicorn/regex-shorthand
  22:21  error  /[ \n\t\r\f(){}:;@!'"\\\][#]|\/(?=\*)/g can be optimized to /[!"#'():;@[\\\]{}]|\/(?=\*)/g  unicorn/regex-shorthand
```

But `/[ \n\t\r\f{}()'"\\;/[\]#]/g` is not equal to `/["#'()/;[\\\]{}]/g` (space symbols were missed).

eslint-plugin-unicorn@14.0.1

**Comment by fisker** [comment]
found minimal reproduction regex `/[ \n\t\r\]]/g`

**Comment by sindresorhus** [comment]
Maybe there's an optimization we can disable? This should also be reported to https://github.com/DmitrySoshnikov/regexp-tree.

**Comment by ai** [comment]
@sindresorhus I like this rule a lot 😍

What do we need to report to `regexp-tree`? Does it generate wrong AST?

**Comment by fisker** [comment]
The result is different now.

```text
  1:1  error  /[ \n\t\r\f"#'()/;[\\\]{}]/g can be optimized to /[\s"#'()/;[\\\]{}]/g                        unicorn/better-regex
  2:1  error  /[ \n\t\r\f(){}:;@!'"\\\][#]|\/(?=\*)/g can be optimized to /[\s!"#'():;@[\\\]{}]|\/(?=\*)/g  unicorn/better-regex
  3:1  error  /[ \n\t\r\]]/g can be optimized to /[\s\]]/g                                                  unicorn/better-regex
```

But seems still not right `[\s]` should equivalent to `[ \f\n\r\t\v\u00a0\u1680\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]` , https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Character_Classes#Types
